### PR TITLE
Import species and breed from excel

### PR DESCRIPTION
**Pull request by Aspodel**

### CODE REVIEW

**Comment by Strypper** [comment]
Need the one for the breeds also

**Comment by Strypper** [comment]
Template file @Aspodel 
[Species.xlsx](https://github.com/Strypper/PetaVerseApi/files/9306421/Species.xlsx)

File: PetaVerseApi/Controller/SpeciesController.cs [hunk 1]
```diff
@@ -115,5 +116,56 @@ public async Task<IActionResult> Delete(int id, CancellationToken cancellationTo
...
+
+            var tempPath = Path.GetTempFileName();
+
+            using (FileStream stream = new FileStream(tempPath, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None, 4096, FileOptions.RandomAccess | FileOptions.DeleteOnClose))
+            {
+                await file.CopyToAsync(stream, cancellationToken);
+
+                System.Text.Encoding.RegisterProvider(System.Text.CodePagesEncodingProvider.Instance);
+
+                using (IExcelDataReader reader = ExcelReaderFactory.CreateReader(stream))
```
Comment (Strypper) [inline PetaVerseApi/Controller/SpeciesController.cs:135]: This should be inside an Excel service

**Comment by Aspodel** [comment]
Done.

**Review by Strypper — APPROVED** [review]
Approved !

**Review by Strypper — APPROVED** [review]
Approved

**Pull request merged**